### DISCUSSION TITLE
输出带格式的推文内容出错

### DISCUSSION THREAD

**Discussion by Swaggeraqua** [zai-org/ChatGLM3#587] [Bad Case]
# 你好，请问我在输入一段文字内容，要求模型生成一段markdown格式的推文后，并保留markdown格式，时常会出现以下问题。
1. 格式混乱，markdown格式没有保留下来
2. 中英文夹杂，甚至全是英文内容，有时会全是繁体字（这个我知道是模型的问题，后续作者会优化）

目前我没有对模型做微调，且超参数也是保持默认值。我设想的解决方法有2个。
1. 调整超参数，因为我生成的推文，属于长文本，想请教一下，该如何调参？
2. 是直接做模型微调，请问微调的数据集该多大，才会有不错的效果？

问题的图示如下
![0](https://github.com/THUDM/ChatGLM3/assets/48743944/975bdbf3-4761-49e1-b492-100310346fd3)
![1](https://github.com/THUDM/ChatGLM3/assets/48743944/c559f165-4ab1-4898-8ac7-1dd1cbd40452)
![2](https://github.com/THUDM/ChatGLM3/assets/48743944/c66f7db3-144a-4f1c-9fd2-16399eec1073)
![3](https://github.com/THUDM/ChatGLM3/assets/48743944/05fe667a-217c-4452-aef0-99e93bcb7e45)

**Comment by zRzRzRzRzRzRzR** [comment]
#300 中有中英文的问题 markdown的话你可以尝试多强调一下

**Reply by zRzRzRzRzRzRzR** [reply]
因为模型对系统的指令跟随效果较好，你试试放在系统提示词
